### ISSUE TITLE
SVector{0}(::Vector) will not construct

### ISSUE THREAD

**Issue by tpapp** [JuliaArrays/StaticArrays.jl#520]
```julia
julia> using StaticArrays

julia> SVector{0}(Int8[])
ERROR: UndefVarError: T not defined
Stacktrace:
 [1] SArray{Tuple{0},T,1,0} where T(::Tuple{}) at /home/tamas/.julia/packages/StaticArrays/Ze5H3/src/SVector.jl:19
 [2] convert at /home/tamas/.julia/packages/StaticArrays/Ze5H3/src/convert.jl:24 [inlined]
 [3] SArray{Tuple{0},T,1,0} where T(::Array{Int8,1}) at /home/tamas/.julia/packages/StaticArrays/Ze5H3/src/convert.jl:5
 [4] top-level scope at none:0

julia> VERSION
v"1.1.0-DEV.480"

(v1) pkg> status
  [90137ffa] StaticArrays v0.8.3
```

**Comment by tpapp** [comment]
Of course the workaround `SVector{0,Int8}(Int8[])` is fine.

**Comment by c42f** [comment]
Will be fixed in #608 — additional tests / comments welcome.
